openupgradelib/openupgrade.py

```diff
@@ -1071,3 +1072,55 @@ def date_to_datetime_tz(
                 AND rp.tz='%(timezone)s';
             """ % values)
     cr.execute("RESET TIMEZONE")
+
+
+def merge_module_names(cr, namespec):
+    """
+    Merge a source module into an existing target model.
+    Can be combined with update_module_names in order to
+    achieve a merge and a simulteneous renaming.
+    Can be used also with the odoo.py --init flag, which should
+    trigger the migration on not-yet-installed new module.
+    Also updates module meta info on loaded data (XML ids).
+    :param namespec: tuple of (source name, target name)
+    """
+    for (old_name, new_name) in namespec:
+        # Get the module id of the old module
+        query = ("SELECT id FROM ir_module_module WHERE name = %s")
+        cr.execute(query, [old_name])
+        id_old = cr.fetchone()[0]
+
+        # Delete meta entries, that might conflict.
+        # They will be recreated by the new module anyhow.
+        query = ("DELETE FROM ir_model_constraint "
+                 "WHERE module = %s")
+        logged_query(cr, query, [id_old])
+        query = ("DELETE FROM ir_model_relation "
+                 "WHERE module = %s")
+        logged_query(cr, query, [id_old])
+        query = ("DELETE FROM ir_translation "
+                 "WHERE module = %s")
```

Comment:
As this library is for all versions, you have to take care the release version, because module field was added on version 7, so as this is executed in pre-init scripts, you should put `if release.version_info[0] > 7:`